src/content/blog/astro-350.mdx

```diff
@@ -0,0 +1,264 @@
+
+## Image optimization improvements
+
+Astro assets has continued to under go improvements and in 3.5 we now have:
+
+- Original images are deleted from the final build that are not used outside of the optimization pipeline. For users with a large number of these images (e.g. thumbnails), this should reduce storage consumption and deployment times.
+- A new property, `propertiesToHash`, has been added to allow specifying which properties of `getImage()` / `<Image />` / `<Picture />` should be used for hashing the result files when doing local transformations. For most services, this will include properties such as `src`, `width` or `quality` that directly changes the content of the generated image.
+- The `<Picture />` component will now use `jpg` and `jpeg` respectively as fallback formats when the original image is in those formats. 
+
+## Integration Hooks to add Middleware
+
+It's now possible in Astro for an integration to add middleware on behalf of the user. Previously when a third party wanted to provide middleware, the user would need to create a `src/middleware.ts` file themselves. Now, adding third-party middleware is as easy as adding a new integration.
+
+For integration authors, there is a new `addMiddleware` function in the `astro:config:setup` hook. This function allows you to specify a middleware module and the order in which it should be applied:
+
+```js
+// my-package/middleware.js
+import { defineMiddleware } from 'astro:middleware';
+
+export const onRequest = defineMiddleware(async (context, next) => {
+  const response = await next();
+
+  if(response.headers.get('content-type') === 'text/html') {
+    let html = await response.text();
+    html = minify(html);
+    return new Response(html, {
+      status: response.status,
+      headers: response.headers
+    });
+  }
+
+  return response;
+});
+```
+
+You can now add your integration's middleware and specify that it runs either before or after the application's own defined middleware (defined in `src/middleware.{js,ts}`)
+
+```js
+// my-package/integration.js
+export function myIntegration() {
+  return {
+    name: 'my-integration',
+    hooks: {
+      'astro:config:setup': ({ addMiddleware }) => {
+        addMiddleware({
+          entrypoint: 'my-package/middleware',
+          order: 'pre'
+        });
+      }
+    }
+  };
+}
+```
+
+## Multiple shiki themes (experimental)
+
+Astro recently upgraded to use the [shikiji](https://github.com/antfu/shikiji) library for syntax highlighting, and we've now added the ability to support multiple themes via a new  `markdown.shikiConfig.experimentalThemes` option.
+
+This enables you to more easily define a light and dark mode theme for syntax highlighted code blocks. You might use it like this:
+
+```js
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  markdown: {
+    shikiConfig: {
+      experimentalThemes: {
+        light: 'github-light',
+				dark: 'github-dark'
```

Comment:
Mix of spaces/tabs here that renders differently in the post.

![image](https://github.com/withastro/astro.build/assets/64310361/691dead5-984e-45f0-bf7b-4b98970c8902)

```suggestion
        light: 'github-light',
        dark: 'github-dark'
```